tests/monotouch-test/NearbyInteraction/NINearbyObjectTest.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright 2020 Microsoft Inc.
+
+#if !MONOMAC
+
+using System;
+using Foundation;
+using ObjCRuntime;
+using NearbyInteraction;
+using NUnit.Framework;
+using OpenTK;
+
+namespace MonoTouchFixtures.NearbyInteraction {
+
+	[TestFixture]
+	[Preserve (AllMembers = true)]
+	public class NINearbyObjectTest {
+
+		[Test]
+		public void DirectionNotAvailable ()
+		{
+			if (!TestRuntime.CheckXcodeVersion (12, 0))
+				Assert.Ignore ("Requires iOS 14.0");
```

Comment:
It's a special case but, since it's all `0` and C# struct are zeroized then the test should _work_ even before iOS 14.

Reply:
I tried running on iOS 13.2 w/o the version check and received the following error:

`2020-08-11 13:55:20.518482-0400 monotouchtest[43690:16037527] 	[FAIL] NINearbyObjectTest.DirectionNotAvailable : System.NullReferenceException : Object reference not set to an instance of an object`

Since NearbyInteraction is a new framework in Xcode12, I think this is expected.

I also tried out the following to double-check:

```
NSCoder coder = new NSCoder ();
NINearbyObject n = new NINearbyObject (coder);
```

It produced this error:
`[FAIL] NINearbyObjectTest.DirectionNotAvailable : System.Exception : Could not create an native instance of the type 'NearbyInteraction.NINearbyObject': the native class hasn't been loaded.`

I believe this is correct behavior since the framework isn't available < iOS 14, lmk if I'm missing something here!

Reply:
```
NINearbyObjectTest.DirectionNotAvailable : System.NullReferenceException : Object reference not set to an instance of an object
```

Reading a field with `dlsym` does not require the field to exists - but maybe it did not like not finding the framework itself. The stack trace would show what cause the NRE.

Anyway not a huge deal, you can leave the version check there :)


```
NSCoder coder = new NSCoder ();
NINearbyObject n = new NINearbyObject (coder);
```

> I believe this is correct behavior since the framework isn't available < iOS 14

Yes. You can't do as it will call code `initWithCoder` on a class (`NINearbyObject`) that has not been loaded into memory.

However the field is different since it's static (no instance has to be created) and return a type (`Vector3`) that already exists. Nothing in your manual code requires the framework to be loaded. In fact it should _work_ (a managed exception is acceptable, not a crash) if someone was to put `NINearbyObject.DirectionNotAvailable` in a debugger watch on an older OS version.

Reply:
That makes sense - if there's no need to instantiate an object to access the field, then I see how it would work.

Someone tried it out on an older simulator and got this stack trace: https://gist.github.com/whitneyschmidt/e5d239d4f83473d29244c7cee86adcf3

Reply:
Stack trace seems incomplete. The first frame is inside nunitlite code, nothing about the test code itself.

Reply:
I have a breakpoint set on `Vector3 vect = NINearbyObject.DirectionNotAvailable;`. When I try and step into it, I end up with the stack trace above 🤔 Is there a better way to try and see what's going on?